cloud/scope/machine.go

```diff
@@ -521,33 +521,29 @@ func (m *MachineScope) ReconcileCreateInstanceOnLB(ctx context.Context) error {
 		backendName := instanceIp + ":" + strconv.Itoa(int(m.OCIClusterAccessor.GetControlPlaneEndpoint().Port))
 		if !m.containsLBBackend(backendSet, backendName) {
 			logger := m.Logger.WithValues("backend-set", *backendSet.Name)
-			workRequest := m.OCIMachine.Status.CreateBackendWorkRequestId
 			logger.Info("Checking work request status for create backend")
-			if workRequest != "" {
-				_, err = ociutil.AwaitLBWorkRequest(ctx, m.LoadBalancerClient, &workRequest)
-				if err != nil {
-					return err
-				}
-			} else {
-				resp, err := m.LoadBalancerClient.CreateBackend(ctx, loadbalancer.CreateBackendRequest{
-					LoadBalancerId: loadbalancerId,
-					BackendSetName: backendSet.Name,
-					CreateBackendDetails: loadbalancer.CreateBackendDetails{
-						IpAddress: common.String(instanceIp),
-						Port:      common.Int(int(m.OCIClusterAccessor.GetControlPlaneEndpoint().Port)),
-					},
-					OpcRetryToken: ociutil.GetOPCRetryToken("%s-%s", "create-backend", string(m.OCIMachine.UID)),
-				})
-				if err != nil {
-					return err
-				}
-				m.OCIMachine.Status.CreateBackendWorkRequestId = *resp.OpcWorkRequestId
-				logger.Info("Add instance to LB backend-set", "WorkRequestId", resp.OpcWorkRequestId)
-				logger.Info("Waiting for LB work request to be complete")
-				_, err = ociutil.AwaitLBWorkRequest(ctx, m.LoadBalancerClient, resp.OpcWorkRequestId)
-				if err != nil {
-					return err
-				}
+			// we always try to create the backend if it exists during a reconcile loop and wait for the work request
+			// to complete. If there is a work request in progress, in the rare case, CAPOCI pod restarts during the
+			// work request, the create backend call may throw an error which is ok, as reconcile will go into
+			// an exponential backoff
+			resp, err := m.LoadBalancerClient.CreateBackend(ctx, loadbalancer.CreateBackendRequest{
+				LoadBalancerId: loadbalancerId,
+				BackendSetName: backendSet.Name,
+				CreateBackendDetails: loadbalancer.CreateBackendDetails{
+					IpAddress: common.String(instanceIp),
+					Port:      common.Int(int(m.OCIClusterAccessor.GetControlPlaneEndpoint().Port)),
+				},
+				OpcRetryToken: ociutil.GetOPCRetryToken("%s-%s", "create-backend", string(m.OCIMachine.UID)),
```

Comment:
This is what makes it idempotent correct?

Reply:
yes